### PR TITLE
added nearby_search + fixed tests

### PR DESCRIPTION
**Pull request by distortia**
First off I waned to thank you for creating this library, it was very easy to extend and get set up.

Second off, this PR is for nearby search of places. The only thing left for places would be text search, which I can handle in another PR. 

Some things of note:

* I've never done doc tests before - I updated some of them to be more reliable and fixed the failures.
* I've never done spec types before either - so please let me know if I messed something up.

Thanks!

### CODE REVIEW

File: lib/google_maps.ex [hunk 1]
```diff
@@ -851,6 +851,112 @@ defmodule GoogleMaps do
     GoogleMaps.get("place/queryautocomplete", params)
   end
 
+@doc """
+  Search for nearby places based on location and keywords
+
+  `nearby_search/3/4` - Most common use for Places API.
```
Comment (sntran) [inline lib/google_maps.ex]: While it's totally fine to use the endpoint name as the function name, I would like to think about using the `place` prefix for these function, i.e., `place_nearby`. This is also to follow the convention of `place_query`.

File: lib/google_maps.ex [hunk 2]
```diff
@@ -851,6 +851,112 @@ defmodule GoogleMaps do
...
+
+  `nearby_search/1/2` - Intended for use with `rankby` 
+    when you cant use radius.
+
+## Args:
+
+    * `location` — The latitude/longitude around which to
```
Comment (sntran) [inline lib/google_maps.ex]: While Google only takes `"latitude,longitude"` stirng as argument, we can support the `coordinate` type defined at the top. The function will adapt accordingly.

File: lib/google_maps.ex [hunk 3]
```diff
@@ -851,6 +851,112 @@ defmodule GoogleMaps do
...
 
+@doc """
+  Search for nearby places based on location and keywords
+
+  `nearby_search/3/4` - Most common use for Places API.
+
+  `nearby_search/1/2` - Intended for use with `rankby` 
+    when you cant use radius.
+
+## Args:
```
Comment (sntran) [inline lib/google_maps.ex]: Please indent this.

File: lib/google_maps.ex [hunk 4]
```diff
@@ -851,6 +851,112 @@ defmodule GoogleMaps do
     GoogleMaps.get("place/queryautocomplete", params)
   end
 
+@doc """
+  Search for nearby places based on location and keywords
+
+  `nearby_search/3/4` - Most common use for Places API.
+
+  `nearby_search/1/2` - Intended for use with `rankby`
```
Comment (sntran) [inline lib/google_maps.ex]: After reading Google's documents, I think we can simply have the function with arity 3, the `coordinate`, `radius`, and the `options`. The `coordinate` and `radius` are required.

Inside the function, we can check for `rankby=distance` option, and ignore the `radius`.

We can get fancy and check for existence of other options when `rankby=distance`, but as a start, we can just assume that the user will check the docs for what options are needed.

File: lib/google_maps.ex [hunk 5]
```diff
@@ -851,6 +851,112 @@ defmodule GoogleMaps do
...
+      Note that radius must not be included if rankby=distance
+      (described under Optional parameters below) is specified
+
+    * `keyword` — The text string on which to search. The Places
+      service will return candidate matches based on this
+      string and order results based on their perceived relevance.
+
+  ## Options:
+
+    * `language` — The language code, indicating in which language the
```
Comment (sntran) [inline lib/google_maps.ex]: `keyword` is also an option, as it's only required when `rankby=distance`.

File: lib/google_maps.ex [hunk 6]
```diff
@@ -851,6 +851,112 @@ defmodule GoogleMaps do
...
+      from 0 (most affordable) to 4 (most expensive), inclusive. 
+      The exact amount indicated by a specific value will vary from 
+      region to region.
+
+    * `opennow` - Returns only those places that are open for business at
+      the time the query is sent. Places that do not specify opening hours
+      in the Google Places database will not be returned if you include
+      this parameter in your query.
+
+    * `region` - The region code, specified as a ccTLD (country code top-level domain)
```
Comment (sntran) [inline lib/google_maps.ex]: I don't see `region` on https://developers.google.com/places/web-service/search#PlaceSearchRequests

File: lib/google_maps.ex [hunk 7]
```diff
@@ -851,6 +851,112 @@ defmodule GoogleMaps do
...
+      * `prominence` - (default). This option sorts results based on their importance.
+        Ranking will favor prominent places within the specified area. 
+        Prominence can be affected by a place's ranking in Google's index,
+        global popularity, and other factors.
+
+      * `distance` - This option biases search results in ascending order by
+        their distance from the specified location. When distance is specified,
+        one or more of keyword, name, or type is required.
+
+  ## Returns
```
Comment (sntran) [inline lib/google_maps.ex]: Please include document for the returned data.

File: lib/google_maps.ex [hunk 8]
```diff
@@ -851,6 +851,112 @@ defmodule GoogleMaps do
...
+  
+  # Search for museums by the white house but rank by distance
+  iex> {:ok, response} = GoogleMaps.nearby_search(
+  ...>  "38.8990252802915,-77.0351808197085",
+  ...>  [rankby: "distance",
+  ...>  keyword: "museum"])
+  iex>  List.first(response["results"])["name"]
+  "Renwick Gallery"
+"""
+  @spec nearby_search(String.t, options() | String.t, integer, String.t options()) :: Response.t()
```
Comment (sntran) [inline lib/google_maps.ex]: Please use only `option()` type.

**Comment by sntran** [comment]
Also, please add your function to the `README`.

**Comment by distortia** [comment]
Thanks for the feedback! Ill address these.

File: lib/google_maps.ex (hunk 6 above)
Comment (distortia) [reply lib/google_maps.ex]: `Region` was under the `textsearch` docs, I just scrolled down too far in their docs 😄 . Good catch!

File: lib/google_maps.ex (hunk 4 above)
Comment (distortia) [reply lib/google_maps.ex]: I struggled a little with how best to approach this. Google said `radius` is required except when `rankby=distance`, which I think is kinda odd on their part. 

Going with arity of 3 and checking for `rankby=distance` makes sense.

**Comment by sntran** [comment]
Let me know when you are at a stopping point for me to review.

**Comment by distortia** [comment]
I am ready

File: lib/google_maps.ex [hunk 9]
```diff
@@ -851,6 +851,197 @@ defmodule GoogleMaps do
...
+    iex> {:ok, response} = GoogleMaps.place_nearby(
+    ...>  "38.8990252802915,-77.0351808197085",
+    ...>  500, 
+    ...>  [rankby: "distance",
+    ...>  keyword: "museum"])
+    iex>  List.first(response["results"])["name"]
+    "Renwick Gallery"
+"""
+  @spec place_nearby(coordinate(), integer, options()) :: Response.t()
+  def place_nearby(location, radius, options \\ []) do
```
Comment (sntran) [inline lib/google_maps.ex]: Can you do another clause to match on a coordinate tuple, i.e., `{latitude, longitude}`. Something like:

```
def place_nearby({latitude, longitude}, radius, options \\ []) do
  place_nearby("#{latitude},#{longitude}", radius, options)
end
```

And of course, check `location` with `is_binary` guard.

This is to accept a full `coordinate` type as argument.

File: lib/google_maps.ex [hunk 10]
```diff
@@ -851,6 +851,197 @@ defmodule GoogleMaps do
...
+
+    * `vicinity` contains a feature name of a nearby location. Often this feature refers to a street or
+      neighborhood within the given results.
+
+    * `permanently_closed` is a boolean flag indicating whether the place has permanently shut down (value true).
+      If the place is not permanently closed, the flag is absent from the response.
+
+  ## Examples
+
+    # Search for museums 500 meters around the White house
```
Comment (sntran) [inline lib/google_maps.ex]: I may be wrong and need to generate the docs to check, but I believe that you do need a full 4 space indentation after `##Examples` for them to show up as code block.

File: lib/google_maps.ex [hunk 11]
```diff
@@ -851,6 +851,197 @@ defmodule GoogleMaps do
...
+  such as: establishments, prominent points of interest,
+  geographic locations, and more. You can search for places
+  either by proximity or a text string. A Place Search 
+  returns a list of places along with summary information
+  about each place; additional information is available
+  via a Place Details query
+
+  ## Args:
+
+    * `location` — The latitude/longitude around which to
```
Comment (sntran) [inline lib/google_maps.ex]: If you decide to support lat/lon tuple besides string, can you update this argument's description?

File: lib/google_maps.ex (hunk 10 above)
Comment (distortia) [reply lib/google_maps.ex]: I'll generate the docs locally and fix this.

File: lib/google_maps.ex (hunk 9 above)
Comment (distortia) [reply lib/google_maps.ex]: Will do!

**Comment by distortia** [comment]
I think I took care of everything we discussed 😄

**Comment by sntran** [comment]
Yes! 👍 . Thank you so much for your patience!

**Comment by distortia** [comment]
No problem, happy to help. I had a lot of fun. I think this may have been my first public PR 😲

**Pull request merged**